firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -374,15 +386,27 @@ public synchronized void onActivityResumed(Activity activity) {
                 + getClassLoadTimeCompat().getDurationMicros(onResumeTime)
                 + " microseconds");
 
-    // Log the app start trace in a non-main thread.
-    executorService.execute(this::logAppStartTrace);
-
     if (!isExperimentTTIDEnabled) {
       // After AppStart trace is logged, we can unregister this callback.
       unregisterActivityLifecycleCallbacks();
     }
   }
 
+  private void dispatchAppStartTrace() {
+    // Log the app start trace in a non-main thread.
+    executorService.execute(this::logAppStartTrace);
+  }
+
+  @Override
+  public void updateSession(PerfSession session) {
+    if (session.sessionId().isEmpty()) {
```

Comment:
When is this true that updateSession is called but the id is null:

Do we need to unregister ourshelves from session updates?